scripts/ci/brotlidec.js

```diff
@@ -0,0 +1 @@
u=u|0;var g=0,B=0,d=0,D=0,v=0,C=0,E=0,H=0,I=0,P=0,F=0,p=0,Y=0,O=0,M=0,G=0,j=0,x=0,N=0,V=0,q=0,W=0;q=(1<<n[A+8>>2])+-16|0;ii(e,o,A,f,i,a);e:do{if(i>>>0>2&f>>>0>127){V=f+-127|0;N=V+i|0;N=N>>>0>f>>>0?f:N;x=o+524292|0;j=o+524296|0;i:while(1){if(V>>>0>=N>>>0)break e;B=n[o>>2]|0;Y=f-V|0;Y=B-(Y>>>0<15?15:Y)|0;O=V&s;M=b+O|0;F=o+4+((ie(t[M>>0]|t[M+1>>0]<<8|t[M+2>>0]<<16|t[M+3>>0]<<24,506832829)|0)>>>15<<2)|0;P=n[F>>2]|0;B=(V&B)<<1;n[F>>2]=V;F=0;d=0;p=0;g=B;B=B|1;G=64;while(1){I=P&s;if((V|0)==(P|0)|(V-P|0)>>>0>Y>>>0|(G|0)==0){a=6;break}H=d>>>0<p>>>0?d:p;if(H>>>0>=129){a=8;break i}C=O+H|0;E=b+(I+128)|0;D=b+(I+124)|0;v=0;a=b+(I+H)|0;while(1){if(a>>>0>D>>>0)break;W=b+(C+v)|0;if((t[a>>0]|t[a+1>>0]<<8|t[a+2>>0]<<16|t[a+3>>0]<<24|0)!=(t[W>>0]|t[W+1>>0]<<8|t[W+2>>0]<<16|t[W+3>>0]<<24|0))break;v=v+4|0;a=a+4|0}while(1){if(a>>>0>=E>>>0)break;if((r[b+(C+v)>>0]|0)!=(r[a>>0]|0))break;v=v+1|0;a=a+1|0}H=H+v|0;f:do{if(H){a=H;C=M;E=b+I|0;while(1){D=r[C>>0]|0;v=r[E>>0]|0;if(D<<24>>24!=v<<24>>24)break;a=a+-1|0;if(!a)break f;else{C=C+1|0;E=E+1|0}}if(D<<24>>24!=v<<24>>24){a=21;break i}}}while(0);if(F){a=23;break i}if(H>>>0>=128){a=25;break}a=n[x>>2]|0;if((t[b+(O+H)>>0]|0)>(t[b+(I+H)>>0]|0)){n[a+(g<<2)>>2]=P;D=(P&n[o>>2])<<1|1;d=H;a=p;g=D;D=(n[x>>2]|0)+(D<<2)|0}else{n[a+(B<<2)>>2]=P;D=(P&n[o>>2])<<1;a=H;B=D;D=(n[x>>2]|0)+(D<<2)|0}F=0;p=a;P=n[D>>2]|0;G=G+-1|0}if((a|0)==6){n[(n[x>>2]|0)+(g<<2)>>2]=n[j>>2];n[(n[x>>2]|0)+(B<<2)>>2]=n[j>>2]}else if((a|0)==25){W=n[x>>2]|0;n[W+(g<<2)>>2]=n[W+((P&n[o>>2])<<1<<2)>>2];W=n[x>>2]|0;n[W+(B<<2)>>2]=n[W+(((P&n[o>>2])<<1|1)<<2)>>2]}V=V+1|0}if((a|0)==8)Be(405873,405905,373,405932);else if((a|0)==21)Be(405955,405905,378,405932);else if((a|0)==23)ue()}}while(0);a=e+8|0;g=vf[n[e>>2]&1](n[a>>2]|0,(i<<4)+16|0)|0;if(!g)ge(1);B=i+1|0;d=0;while(1){if((d|0)==(B|0))break;n[g+(d<<4)>>2]=1;n[g+(d<<4)+4>>2]=0;n[g+(d<<4)+8>>2]=0;l[g+(d<<4)+12>>2]=Q;d=d+1|0}W=di(e,i,f,b,s,A,q,w,o,g)|0;n[c>>2]=(n[c>>2]|0)+W;Bi(i,f,q,g,w,k,h,u);Qf[n[e+4>>2]&1](n[a>>2]|0,g);return}function Ci(e,i,f,b,s,A,w,h,c,u,g,B,d){e=e|0;i=i|0;f=f|0;b=b|0;s=s|0;A=A|0;w=w|0;h=h|0;c=c|0;u=u|0;g=g|0;B=B|0;d=d|0;var D=0,v=0,C=0,E=0.0,H=0,I=0,P=0,F=0,p=0,Y=0,O=0,M=0,G=0,j=0,x=0,N=0,V=0,q=0,W=0,y=0,U=0,m=0,T=0,X=0,R=0,z=0,J=0,K=0,L=0,S=0,Z=0,_=0,$=0,ee=0,fe=0,re=0,ae=0,ne=0,te=0,be=0,se=0,le=0,Ae=0,oe=0,we=0,ke=0,he=0,ce=0,de=0,De=0,Qe=0.0;de=o;o=o+12256|0;ne=de+12028|0;ae=de+12024|0;ke=de+11e3|0;oe=de+8184|0;we=de+6104|0;Ae=de+5080|0;$=de+4928|0;ce=de+4912|0;he=de;te=(1<<n[w+8>>2])+-16|0;re=i<<2;se=e+8|0;le=vf[n[e>>2]&1](n[se>>2]|0,re)|0;if(!le)ge(1);_=i>>>0>127?f+i+-127|0:f;fe=vf[n[e>>2]&1](n[se>>2]|0,i<<5)|0;m=fe;if(!fe)ge(1);ii(e,h,w,f,i,b);e:do{if(i>>>0>2&f>>>0>127){U=f+-127|0;y=U+i|0;y=y>>>0>f>>>0?f:y;q=h+524292|0;W=h+524296|0;i:while(1){if(U>>>0>=y>>>0)break e;v=n[h>>2]|0;j=f-U|0;j=v-(j>>>0<15?15:j)|0;x=U&A;N=s+x|0;M=h+4+((ie(t[N>>0]|t[N+1>>0]<<8|t[N+2>>0]<<16|t[N+3>>0]<<24,506832829)|0)>>>15<<2)|0;O=n[M>>2]|0;v=(U&v)<<1;n[M>>2]=U;M=0;C=0;G=0;D=v;v=v|1;V=64;while(1){Y=O&A;if((U|0)==(O|0)|(U
```

Comment:
What is the source/license of this?

Reply:
I actually just used an old version of this that I had lying around locally :-D. I *think* it's https://github.com/yisibl/brotli-js, so MIT. But if we go this way, I'd replace it by a proper rebuild anyways.

Reply:
Actually, I think the file even contains compression *and* decompression logic, so we only actually need half of it.